cmd/ocitools/validate.go

```diff
@@ -397,6 +398,26 @@ func checkLinuxResources(r rspec.Resources, hostCheck bool) (msgs []string) {
 			msgs = append(msgs, fmt.Sprintf("Minimum memory limit should be larger than memory reservation"))
 		}
 	}
+	if r.Network != nil && hostCheck {
+		var exist bool
+		interfaces, err := net.Interfaces()
+		if err != nil {
+			msgs = append(msgs, err.Error())
+			return
+		}
+		for _, prio := range r.Network.Priorities {
+			exist = false
+			for _, ni := range interfaces {
+				if prio.Name == ni.Name {
+					exist = true
+					break
+				}
+			}
+			if !exist {
+				msgs = append(msgs, fmt.Sprintf("Interface %s does not exist on host", prio.Name))
```

Comment:
On Wed, Aug 24, 2016 at 06:36:35PM -0700, Ma Shimiao wrote:
> we restrict to execute the check when container needs a new network
> namespace or does not set network namespace.

It's not clear to me when cgroups are created/joined/configured
vs. namespaces being joined/created.  I'm also not clear on how the
cgroups are associated with network namespaces (do all network cgroups
limit root-network-namespace interfaces?  Is there a way to bind
network cgroups to a particular non-root network namespace?).  So this
is all very murky for me.

I'd rather:

1. Wait until opencontainers/runtime-spec#537 and
   opencontainers/runtime-spec#538 settle down.

2. Follow them up with language to replace the current
   namespace-specific no-tweaking language with some more generic
   no-tweaking language ((b) in [1]).

3. Have somebody research and report back on how network cgroups and
   network namespaces are supposed to work together.

and then come back and figure out how to handle validation.  Before
those pieces are in place, I think the link between the spec and the
validating code will be a bit hand-wavy.  If we want to land some
hand-wavy stuff because we expect that's where the spec will end up,
that's fine; I just don't have a clear enough understanding
personally.

[1]: https://github.com/opencontainers/ocitools/pull/200/files/df860e11435241582fafd1fbd47649cdf4823ba6#r76095670